api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -2676,4 +2685,125 @@ public void getEncounterRolesByName_shouldFindEncounterRolesByName() throws Exce
 		assertEquals(encounterRoles.size(), 1);
 		assertEquals(encounterRoles.get(0).getName(), name);
 	}
+	
+
+	@Test
+	public void shouldSaveOrderGroupAlongWithOrders() throws Exception {
+		executeDataSet(ORDER_SET);
+		
+		//Created a new Encounter
+		Encounter encounter = new Encounter();
+		encounter.setPatient(Context.getPatientService().getPatient(3));
+		encounter.setEncounterType(Context.getEncounterService().getEncounterType(1));
+		encounter.setEncounterDatetime(new Date());
+		
+		Context.getEncounterService().saveEncounter(encounter);
+		
+		Integer encounterId = Context.getEncounterService().getEncounterByUuid(encounter.getUuid()).getId();
+		
+		//Created a new OrderGroup
+		OrderSet orderSet = Context.getOrderSetService().getOrderSetById(2000);
+		OrderGroup orderGroup = new OrderGroup();
+		orderGroup.setOrderSet(orderSet);
+		orderGroup.setEncounter(Context.getEncounterService().getEncounter(encounterId));
+		
+		//Added this OrderGroup to two new orders
+		Order newOrder1 = new OrderBuilder().withAction(Order.Action.NEW).withPatient(3).withConcept(1000)
+		        .withCareSetting(1).withOrderer(1).withEncounter(encounterId).withDateActivated(new Date()).withOrderType(17)
+		        .withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+		newOrder1.setOrderGroup(orderGroup);
+		
+		Order newOrder2 = new OrderBuilder().withAction(Order.Action.NEW).withPatient(3).withConcept(1001)
+				.withCareSetting(1).withOrderer(1).withEncounter(encounterId).withDateActivated(new Date()).withOrderType(17)
+				.withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+		newOrder2.setOrderGroup(orderGroup);
+		
+		//Add these orders to the Encounter
+		encounter.addOrder(newOrder1);
+		encounter.addOrder(newOrder2);
+		
+		Context.getEncounterService().saveEncounter(encounter);
+		
+		Context.flushSession();
+		
+		
+		List<Order> orders = new ArrayList<Order>();
+		orders.addAll(Context.getEncounterService().getEncounterByUuid(encounter.getUuid()).getOrders());
+		
+		assertEquals("Only one orderGroup gets saved for both the orders ",orders.get(0).getOrderGroup().getId(), orders.get(1).getOrderGroup().getId());
```

Comment:
We just need to compare the two orderGroups. It has equals function defined. Not necessary to check ids.